src/table/boot.rs

```diff
@@ -481,10 +493,15 @@ impl BootServices {
     /// This function attempts to get the protocol implementation of a handle,
     /// based on the protocol GUID.
     ///
+    /// It is recommended that all new drivers and applications use
+    /// [`open_protocol`] instead of `handle_protocol`.
```

Comment:
Should we mark the `handle_protocol` method as deprecated in view of your findings, and transition all of the usages within our repo to `open_protocol` instead of `handle_protocol`?

Reply:
I think that makes sense, yes. Rebased and added another commit on top to convert all remaining uses of `handle_protocol`.

Reply:
Thanks. Could we also include [a message](https://doc.rust-lang.org/reference/attributes/diagnostics.html#the-deprecated-attribute) with a deprecation notice, such as ```#[deprecated(note = "it is recommended to use `open_protocol` instead")]```?

Reply:
Done